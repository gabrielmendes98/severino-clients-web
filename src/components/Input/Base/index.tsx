import TextField, { TextFieldProps } from '@mui/material/TextField';

export type InputBaseProps = TextFieldProps & {};

const InputBase = ({ children, size = 'small', ...props }: InputBaseProps) => (
  <TextField size={size} {...props}>
    {children}
  </TextField>
);

export default InputBase;
