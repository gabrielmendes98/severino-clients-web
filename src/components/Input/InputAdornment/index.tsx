import MuiInputAdornment, {
  InputAdornmentProps,
} from '@mui/material/InputAdornment';

const InputAdornment = ({ children, ...props }: InputAdornmentProps) => (
  <MuiInputAdornment {...props}>{children}</MuiInputAdornment>
);

export default InputAdornment;
