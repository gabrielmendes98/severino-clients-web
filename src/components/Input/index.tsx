import { useField } from 'formik';
import InputBase, { InputBaseProps } from './Base';

type Props = InputBaseProps & {
  name: string;
};

const Input = ({ name, children, ...props }: Props) => {
  const [field, meta] = useField(name);
  const { error, touched, ...restMeta } = meta;

  const hasError = Boolean(touched && error);

  return (
    <InputBase
      error={hasError}
      helperText={hasError && error}
      {...restMeta}
      {...field}
      {...props}
    >
      {children}
    </InputBase>
  );
};

export default Input;
