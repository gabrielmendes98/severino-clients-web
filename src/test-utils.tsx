import React, { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { PreloadedState } from '@reduxjs/toolkit';
import ThemeProvider from './common/providers/Theme';
import { AppStore, RootState, setupStore } from './common/store/store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

const customRender = (
  ui: ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) => {
  const Wrapper: FC<{ children: ReactNode }> = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider>{children}</ThemeProvider>
    </Provider>
  );

  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

/* eslint-disable */
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { customRender as render };
/* eslint-enable */
