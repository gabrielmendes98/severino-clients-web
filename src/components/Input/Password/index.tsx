import VisibilityOffIcon from '@mui/icons-material/VisibilityOff';
import VisibilityIcon from '@mui/icons-material/Visibility';
import { useState } from 'react';
import IconButton from 'components/IconButton';
import Input, { InputProps } from '../index';
import InputAdornment from '../InputAdornment';

const PasswordInput = ({ children, InputProps, ...props }: InputProps) => {
  const [visible, setVisible] = useState(false);

  const preventMouseEvent = (event: React.MouseEvent<HTMLButtonElement>) => {
    event.preventDefault();
  };

  const toggleVisibility = () => setVisible(init => !init);

  return (
    <Input
      {...props}
      type={visible ? 'text' : 'password'}
      InputProps={{
        ...InputProps,
        endAdornment: (
          <InputAdornment position="end">
            <IconButton
              aria-label="mostrar ou esconder senha"
              onClick={toggleVisibility}
              onMouseDown={preventMouseEvent}
              onMouseUp={preventMouseEvent}
              edge="end"
            >
              {visible ? <VisibilityIcon /> : <VisibilityOffIcon />}
            </IconButton>
          </InputAdornment>
        ),
      }}
    >
      {children}
    </Input>
  );
};

export default PasswordInput;
