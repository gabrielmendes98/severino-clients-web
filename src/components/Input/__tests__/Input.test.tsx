import { Form, Formik } from 'formik';
import { render, waitFor, screen, userEvent } from 'test-utils';
import Input from '../index';

it('should match snapshot', () => {
  const { container } = render(
    <Formik initialValues={{ test: '' }} onSubmit={jest.fn()}>
      <Form>
        <Input name="test" label="test input" />
        <button type="submit">submit</button>
      </Form>
    </Formik>,
  );
  expect(container).toMatchSnapshot();
});

it('should update form values', async () => {
  const onSubmit = jest.fn();
  render(
    <Formik initialValues={{ test: '' }} onSubmit={onSubmit}>
      <Form>
        <Input name="test" label="test input" />
        <button type="submit">submit</button>
      </Form>
    </Formik>,
  );

  await userEvent.type(screen.getByLabelText('test input'), 'lalala');
  userEvent.click(screen.getByRole('button', { name: 'submit' }));

  await waitFor(() => {
    expect(onSubmit).toHaveBeenCalledWith(
      {
        test: 'lalala',
      },
      expect.any(Object),
    );
  });
});
