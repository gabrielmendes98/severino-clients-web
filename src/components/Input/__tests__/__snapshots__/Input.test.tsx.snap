// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="Toastify"
  />
  <form
    action="#"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-10ki2om-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":r0:"
        id=":r0:-label"
      >
        test input
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-78i8ym-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          name="test"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1ftyaf0"
          >
            <span>
              test input
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      type="submit"
    >
      submit
    </button>
  </form>
</div>
`;
